patches/server/0965-Make-banners-modifiable-with-both-BannerMeta-and-Blo.patch

```diff
@@ -0,0 +1,270 @@
++            return;
++        }
++        // Paper end - Use underlying block state
+         this.base = color;
+     }
+ 
+     @Override
+     public List<Pattern> getPatterns() {
++        if (true) return this.bannerBlock.getPatterns(); // Paper - Use underlying block state
+         return new ArrayList<Pattern>(this.patterns);
+     }
+ 
+     @Override
+     public void setPatterns(List<Pattern> patterns) {
++        // Paper start - Use underlying block state
++        if (true) {
++            this.bannerBlock.setPatterns(patterns);
++            this.setBlockState(this.bannerBlock); // Send changes to the underlying class
++            return;
++        }
++        // Paper end - Use underlying block state
+         this.patterns = new ArrayList<Pattern>(patterns);
+     }
+ 
+     @Override
+     public void addPattern(Pattern pattern) {
++        // Paper start - Use underlying block state
++        if (true) {
++            this.bannerBlock.addPattern(pattern);
++            this.setBlockState(this.bannerBlock); // Send changes to the underlying class
++            return;
++        }
++        // Paper end - Use underlying block state
+         this.patterns.add(pattern);
+     }
+ 
+     @Override
+     public Pattern getPattern(int i) {
++        if (true) return this.bannerBlock.getPattern(i); // Paper - Use underlying block state
+         return this.patterns.get(i);
+     }
+ 
+     @Override
+     public Pattern removePattern(int i) {
++        // Paper start - Use underlying block state
++        if (true) {
++            Pattern removedPattern = this.bannerBlock.removePattern(i);
++            this.setBlockState(this.bannerBlock); // Send changes to the underlying class
++            return removedPattern;
++        }
++        // Paper end - Use underlying block state
+         return this.patterns.remove(i);
+     }
+ 
+     @Override
+     public void setPattern(int i, Pattern pattern) {
++        // Paper start - Use underlying block state
++        if (true) {
++            this.bannerBlock.setPattern(i, pattern);
++            this.setBlockState(this.bannerBlock); // Send changes to the underlying class
++            return;
++        }
++        // Paper end - Use underlying block state
+         this.patterns.set(i, pattern);
+     }
+ 
+     @Override
+     public int numberOfPatterns() {
++        if (true) return this.bannerBlock.numberOfPatterns(); // Paper - Use underlying block state
+         return this.patterns.size();
+     }
+ 
+@@ -191,6 +261,7 @@ public class CraftMetaBanner extends CraftMetaItem implements BannerMeta {
+     ImmutableMap.Builder<String, Object> serialize(ImmutableMap.Builder<String, Object> builder) {
+         super.serialize(builder);
+ 
++        if (true) return builder; // Paper - Let CraftMetaBlockState serialize for us
+         if (this.base != null) {
+             builder.put(BASE.BUKKIT, this.base.toString());
+         }
```

Comment:
This isn't a request for change necessarily, just a note that changing this will make items serialized on paper not be de serialized correctly if someone goes backwards to use spigot. I am fine with it however.